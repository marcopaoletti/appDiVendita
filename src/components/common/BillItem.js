import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';


const BillItem = ({ item, removeItem }) => {
  return (
    <TouchableOpacity style={styles.row} onPress={() => removeItem()}>
      <Text style={styles.productName}>{item.name}</Text >
      <Text style={styles.productPrice}>{`â‚¬ ${item.price.toFixed(2)}`}</Text>
    </TouchableOpacity>
  );
};

BillItem.propTypes = {
  item: PropTypes.shape().isRequired,
  removeItem: PropTypes.func.isRequired,
};

const styles = {
  row: {
    width: '100%',
    paddingHorizontal: 28,
    flexDirection: 'row',
    justifyContent: 'space-between',
    height: 40,
    alignItems: 'center',
  },
  productName: {
    fontSize: 15,
  },
  productPrice: {
    fontSize: 15,
    color: 'grey',
  },
};


export { BillItem };
