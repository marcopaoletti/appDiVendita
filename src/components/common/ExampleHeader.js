import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

const ExampleHeader = ({ headerText }) => {
  return (
    <View>
      <Text>{headerText}</Text>
    </View>
  );
};

ExampleHeader.propTypes = {
  headerText: PropTypes.string.isRequired,
};

export { ExampleHeader };
