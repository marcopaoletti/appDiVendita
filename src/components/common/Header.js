import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { Icon } from 'react-native-elements';

import { propWidth } from 'Utils';

const Header = ({ title, icon, navigate }) => {
  return (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <Text style={styles.text}> {title} </Text>
      </View>
      <TouchableOpacity style={styles.iconContainer} onPress={() => navigate()}>
        <Icon
          name={icon}
          type="material-community"
          style={styles.icon}
        />
      </TouchableOpacity>
    </View>
  );
};

Header.propTypes = {
  title: PropTypes.string,
  navigate: PropTypes.func.isRequired,
  icon: PropTypes.string.isRequired,
};

Header.defaultProps = {
  title: 'App di vendita',
};

const styles = {
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: propWidth(10),
  },
  textContainer: {
  },
  text: {
    fontSize: 25,
  },
  iconContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
};

export { Header };
