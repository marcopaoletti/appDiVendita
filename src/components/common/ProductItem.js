import React from 'react';
import { Image, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { Card } from 'react-native-elements';


const ProductItem = ({ name, price, img, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <Card title={`${name}, ${price} euro`}>
        <Image
          style={{ width: '100%', height: 150 }}
          source={{ uri: img }}
        />
      </Card>
    </TouchableOpacity>
  );
};

ProductItem.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  img: PropTypes.string.isRequired,
  onPress: PropTypes.func,
};

ProductItem.defaultProps = {
  onPress: null,
};

export { ProductItem };
