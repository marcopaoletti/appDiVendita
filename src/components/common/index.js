export * from 'common/ExampleHeader';
export * from 'common/ProductItem';
export * from 'common/BillItem';
export * from 'common/Header';
export { default as Cashier } from 'common/Cashier';
