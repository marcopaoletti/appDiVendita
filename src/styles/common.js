export const EXAMPLE_COLOR = '#8aa7ad';
