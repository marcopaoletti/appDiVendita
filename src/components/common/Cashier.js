import React, { Component } from 'react';
import { View, Text, FlatList } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { BillItem } from 'common';
import { resetBill } from 'actions';

class Cashier extends Component {
  render() {
    return (
      <View style={styles.cassa}>
        <View style={styles.billContainer}>
          <FlatList
            data={this.props.billItems}
            renderItem={({ item }) => <BillItem item={item} />}
            ListEmptyComponent={() =>
              <Text style={styles.emptyItem}>Non ci sono elementi nel carrello</Text>
            }
            keyExtractor={(item, index) => index.toString()}
          />
        </View>
        <View style={styles.buttonContainer}>
          <Button
            small
            raised
            icon={{ name: 'squirrel', type: 'octicon' }}
            title={`Paga ${this.props.amount.toFixed(2)} euro`}
            onPress={() => this.props.resetBill()}
          />
        </View>
      </View>
    );
  }
}

Cashier.propTypes = {
  billItems: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  resetBill: PropTypes.func.isRequired,
  amount: PropTypes.number,
};

Cashier.defaultProps = {
  amount: 0,
};

const styles = {
  cassa: {
    flex: 0.3,
    paddingTop: 15,
  },
  billContainer: {
    flex: 0.8,
  },
  buttonContainer: {
    flex: 0.2,
  },
  emptyItem: {
    flex: 1,
    textAlign: 'center',
  },
};

const mapStatetoProps = ({ bill }) => ({
  billItems: bill.list,
  amount: bill.list.reduce((total, product) => total + product.price, 0),
});

export default connect(mapStatetoProps, {
  resetBill,
})(Cashier);
