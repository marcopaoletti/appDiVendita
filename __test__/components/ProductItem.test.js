import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRender from 'react-test-renderer/shallow';

import { ProductItem } from 'common';

describe('<ProductItem>', () => {
  const onPressMock = jest.fn();

  const props = {
    name: 'Super Prodotto',
    price: 11.45,
    img: 'https://superimmagine.com',
    onPress: onPressMock,
  };

  it('should render without crashing', () => {
    const rendered = renderer.create(<ProductItem {...props} />).toJSON();
    expect(rendered).toBeTruthy();
  });

  const shallowRenderer = new ShallowRender();

  it('should render properly even without a name', () => {
    const newProps = { ...props };
    newProps.name = '';
    expect(shallowRenderer.render(<ProductItem {...newProps} />)).toMatchSnapshot();
  });

  it('clicking on the card should call the proper function', () => {
    const rendered = renderer.create(<ProductItem {...props} />).root;
    rendered.props.onPress();
    expect(onPressMock).toBeCalled();
  });

});
