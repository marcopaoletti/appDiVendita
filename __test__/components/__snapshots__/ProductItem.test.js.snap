// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem> should render properly even without a name 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <Card
    title=", 11.45 euro"
  >
    <Image
      source={
        Object {
          "uri": "https://superimmagine.com",
        }
      }
      style={
        Object {
          "height": 150,
          "width": "100%",
        }
      }
    />
  </Card>
</TouchableOpacity>
`;
